ui/client/reducers/nodeDetailsState.ts

```diff
@@ -1,9 +1,10 @@
 import {Action} from "../actions/reduxTypes"
 import {ValidationData} from "../actions/nk"
-import {NodeValidationError, UIParameter} from "../types"
+import {NodeValidationError, UIParameter, UITypedExpression} from "../types"
 
 export type NodeDetailsState = {
-    parameters? : Record<string, UIParameter>,
+    parameters? : UIParameter[],
+    typedExpressions?: UITypedExpression[],
```

Comment:
Do we want to show typing results only when user is changing expressions? Or also when they open node modal? If so, we need to pass typedExpressions also in ValidationResult?

Reply:
Good point, I think it would be nice to display type info when modal is opened and no changes yet applied,
- will add type info to ValidatinResult

Reply:
@mproch 
https://github.com/TouK/nussknacker/pull/1191/commits/d0abd3e3ead301d1a1b983ec531cf9ddb4025632

Reply:
@mproch 
It turns out, that TypingResult is used in non-typed js files (non TS),
so I've changed type definition to algebraic type:
https://github.com/TouK/nussknacker/pull/1191/commits/0ee10e1583ce0d0c3562ae268b08d4d45bc791c1

Also TS, has very nice support for narrowing union types in if esle structures and type guard, so it won't be too much hassle to handle it.